### ISSUE TITLE
[Broken Website] Messenger.com with gradiant thème

### ISSUE THREAD

**Issue by DrumSlayers** [darkreader/darkreader#4438]
**Website**
The link to the broken website
https://www.messenger.com/ (desktop)

**How to Reproduce**
An example of this is
>
> 1. Go to Messenger and login
> 2. Make sure you are on the latest Messenger CSS Revision (clear cache)
> 3. Use a gradiant thème in your conversation (like this \/ )
![image](https://user-images.githubusercontent.com/4050278/101546071-8561cc00-39a8-11eb-8c5a-1d91984bfacf.png)
> 4. Send a messsage to someone
> 5. See that the message is surrounded by weird colors

**Expected behavior**
Sended messages look like received messages and same in every theme

**Actual behavior**
Sended messages have weird lookin corners, when received messages don't. But it only happens in gradiant themes

**Screenshots**
Glitched text
![image](https://user-images.githubusercontent.com/4050278/101546127-9f031380-39a8-11eb-9dea-3c5ee25304e9.png)

Normal text
![image](https://user-images.githubusercontent.com/4050278/101546149-a75b4e80-39a8-11eb-9f15-06202e03dabf.png)

**System info:**

- OS: Windows 10 20H2
- Browser: Chrome Version 87.0.4280.88
- Darkreader Version: 4.9.26

**Additional context**
This issue is happening since the new revision of Messenger theme/look. Happenned on 08/12/2020 for me.

**Comment by Gusted** [comment]
@Myshor :D

**Comment by Myshor** [comment]
quick fix for dark mode in dark reader made in #4439
(but it's not working fine if someone will use Dark Reader's light mode - but I'm out of time and ideas how to use different colors when $ can not be used easily for fix)

**Comment by SebastianRasch** [comment]
Also, avatars are way too dark since the update. "--darkreader-inline-fill" is wrong for these; should be bright, not dark. Can't change it in the editor since the Dark Reader rules already have a higher priority. Please fix! 🙏🏻

**Comment by Myshor** [comment]
I was fixing dark avatars already.
Check and compare newest messenger.com section in https://github.com/darkreader/darkreader/blob/master/src/config/dynamic-theme-fixes.config

**Comment by SebastianRasch** [comment]
OK nice, that's gonna be in 4.10 I guess?

**Comment by Myshor** [comment]
In next release to be sure. 😀 but you can always copy that part and paste into developer tools in extension to use it earlier. 😉

**Comment by SebastianRasch** [comment]
Yeah I just did that actually 😉 Works like a charm, thanks!

Only video controls are still a bit difficult to see but that's a minor problem.

**Comment by Myshor** [comment]
Yeah, video controls are bit problematic, cause when I try to apply a fix for them they dissapear completely.
But when I will find solution for them I will add that too. :)

Alternatively you can use messenger from facebook. https://www.facebook.com/messages
There the video controls works fine. ;)

As I don't have much time to do fixes in the week then the weekends are the time for deeper analysis. I'll check what I can do with that controls when ussing messenger.com

**Comment by Myshor** [comment]
Ok. I made quick fix for videos here => #4527 

I hope facebook won't change class for that buttons or the imageatlas with buttons too quick. They like making changes in the code all the time.

**Comment by SebastianRasch** [comment]
Super nice, thank you! Looking good!

**Comment by SMotaal** [comment]
@Myshor it seems that they rolled out a style change that broke the contrast on incoming messages in dynamic mode:

<img width="558" alt="image" src="https://user-images.githubusercontent.com/849613/129546082-092d2b9b-acd4-4568-9f60-5fa43792f178.png">

In Facebook's own dark mode:

<img width="555" alt="image" src="https://user-images.githubusercontent.com/849613/129546315-829224e1-f715-4055-852a-390cc759ea2a.png">

Thanks for this BTW:

> Alternatively you can use messenger from facebook. https://www.facebook.com/messages

I'm sure folks here can relate to times when accessibility feels like a sinking ship with never-ending sources of friction, and this just made my day ❤️

**Comment by Myshor** [comment]
@SMotaal I see it's some bot. I do not have any in my "contacts" to test. Maybe you give some tip to find this.

**Comment by SMotaal** [comment]
@Myshor the bot was simply ideal to take the screenshots, but the problem carries in all conversations on my end.

Here's a redacted screenshot for dynamic mode which is best suited:

<img width="1280" alt="Screen Shot 2021-08-16 DarkReader A" src="https://user-images.githubusercontent.com/849613/129594729-1c02156a-b9c6-49e4-b68f-d2c42c600364.png">

I also tried with different brightness and contrast settings but no luck.

Also another for filter mode which is hardly unsuited:

<img width="1280" alt="Screen Shot 2021-08-16 DarkReader B" src="https://user-images.githubusercontent.com/849613/129594765-c8b1052c-23e3-403d-9cc3-7463ce45a25f.png">

The culprit seems to be:

```html
<div 
  role="none"
  class="l60d2q6s d1544ag0 sj5x9vvc tw6a2znq l9j0dhe7 ni8dbmo4 stjgntxs e72ty7fz qlfml3jp inkptoze qmr60zad jm1wdb64 qv66sw1b ljqsnud1 odn2s2vf tkr6xdv7"
  style="background-color: var(--wash);"
/>
```

Where I believe the inline style attribute `background-color: var(--wash);` ends up overriding DarkReader's injected styles.

Hope this helps 🖖

**Comment by Myshor** [comment]
![image](https://user-images.githubusercontent.com/56877029/129600608-068cf399-dedc-4a83-890f-7bfddb001959.png)

It works fine for me without any more fixes. And I even checked for new messages recieved. Still dark incoming.

**Comment by Myshor** [comment]
One more question: What browser are you using?

**Comment by SMotaal** [comment]
> One more question: What browser are you using?

Safari

**Comment by Myshor** [comment]
Safari version has different code base and will need some more analysis in our team.
